File: assignment/A3TaTanPhong_SE18D04_FE/src/services/authService.js
import api from "./api";

export const authService = {
  // Đăng ký tài khoản (RegisterRequest)
  register: async (registerRequest) => {
    const response = await api.post("/auth/register", registerRequest);
    return response.data; // Trả về ApiResponse<AuthResponse>
  },

  // Đăng nhập (LoginRequest)
  login: async (loginRequest) => {
    const response = await api.post("/auth/login", loginRequest);
    return response.data; // Trả về ApiResponse<AuthResponse>
  },
};

File: assignment/A3TaTanPhong_SE18D04_BE/src/main/java/fu/se/sba301/phongtt/a3tatanphong_se18d04/services/AuthService.java
package fu.se.sba301.phongtt.a3tatanphong_se18d04.services;

import fu.se.sba301.phongtt.a3tatanphong_se18d04.dto.request.LoginRequest;
import fu.se.sba301.phongtt.a3tatanphong_se18d04.dto.request.RegisterRequest;
import fu.se.sba301.phongtt.a3tatanphong_se18d04.dto.response.AuthResponse;

public interface AuthService {
    AuthResponse register(RegisterRequest request);
    AuthResponse login(LoginRequest request);
}